cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/SearchCommand.java

```diff
@@ -0,0 +1,126 @@
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.cli.cmd;
+
+import org.ballerinalang.central.client.CentralAPIClient;
+import org.ballerinalang.central.client.model.PackageSearchJsonSchema;
+import org.ballerinalang.cli.module.exeptions.CommandException;
+import org.ballerinalang.tool.BLauncherCmd;
+import org.wso2.ballerinalang.util.RepoUtils;
+import picocli.CommandLine;
+
+import java.io.PrintStream;
+import java.util.List;
+
+import static io.ballerina.cli.cmd.Constants.SEARCH_COMMAND;
+import static io.ballerina.cli.utils.PrintUtils.printPackages;
+import static io.ballerina.runtime.util.RuntimeConstants.SYSTEM_PROP_BAL_DEBUG;
+import static org.ballerinalang.tool.LauncherUtils.createUsageExceptionWithHelp;
+
+/**
+ * This class represents the "ballerina search" command.
+ *
+ * @since 2.0.0
+ */
+@CommandLine.Command(name = SEARCH_COMMAND, description = "search for modules within Ballerina Central")
+public class SearchCommand implements BLauncherCmd {
+    private static PrintStream outStream = System.err;
+
+    @CommandLine.Parameters
+    private List<String> argList;
+
+    @CommandLine.Option(names = {"--help", "-h"}, hidden = true)
+    private boolean helpFlag;
+
+    @CommandLine.Option(names = "--debug", hidden = true)
+    private String debugPort;
+
+    @Override
+    public void execute() {
+        if (helpFlag) {
+            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(SEARCH_COMMAND);
+            outStream.println(commandUsageInfo);
+            return;
+        }
+
+        if (null != debugPort) {
+            System.setProperty(SYSTEM_PROP_BAL_DEBUG, debugPort);
+        }
+
+        if (argList == null || argList.isEmpty()) {
+            throw createUsageExceptionWithHelp("no keyword given");
+        }
+
+        if (argList.size() > 1) {
+            throw createUsageExceptionWithHelp("too many arguments");
+        }
+
+        String searchArgs = argList.get(0);
+        searchInCentral(searchArgs);
+        Runtime.getRuntime().exit(0);
+    }
+
+    @Override
+    public String getName() {
+        return SEARCH_COMMAND;
+    }
+
+    @Override
+    public void printLongDesc(StringBuilder out) {
+        out.append("searches for packages within Ballerina Central \n");
+    }
+
+    @Override
+    public void printUsage(StringBuilder out) {
+        out.append(" ballerina search [<org>|<package>|<text>] \n");
+    }
+
+    @Override
+    public void setParentCmdParser(CommandLine parentCmdParser) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Search for packages in central.
+     *
+     * @param query search keyword.
+     */
+    private static void searchInCentral(String query) {
+        try {
+            CentralAPIClient client = new CentralAPIClient();
+            PackageSearchJsonSchema packageSearchJsonSchema = client.searchPackage(query);
```

Comment:
Why the model name has Schema, can we call it PackageSearchResult etc

Reply:
renamed model classes